### PR TITLE
Feature/3 isolate individual tests

### LINKED ISSUE #3: Isolate individual tests and parallel test runs
Make sure the tests are self-sufficient and encapsulated, not affecting each other. Also when several people/pipelines are running the tests in parallel, these runs should also not affect each other.

Clear/restore the shared state after each `it` or `describe`.

E.g. if funds are sent, they should be returned back.

**Stretch goal:**
Create a pool of 100 safes with preloaded owners and funds, and pick one for each suite run. E.g. pick a safe based on the timestamp % 100.

### PR DESCRIPTION
**Pull request by juampibermani**
## What it solves

Resolves #3 

## How this PR fixes it

Each test does not depend anymore on other tests
When one test suite finishes we check if there are pending tx's, if there're, we cancel them

### CODE REVIEW

**Review by katspaugh — APPROVED** [review]
Nice!

Does it fully isolate the tests though? I thought the main problem was that individual `.test.js` files were mutating shared information (funds, transaction list).

**Review by dasanra — COMMENTED** [review]
I can't see how this changes solve #3

Also I'm not sure that removing the steps we split the tests is something good. Now in case of failure it will be really hard to find where is the test failing and so check for errors in that code section

File: src/add_remove_owner.test.js [hunk 1]
```diff
@@ -39,10 +39,11 @@ describe('Adding and removing owners', () => {
   const newOwnerAddress = NON_OWNER_ADDRESS
   let currentNonce = ''
 
-  test('Filling Form and submitting tx', async (done) => {
+  test('Add and remove an owner', async (done) => {
```
Comment (katspaugh) [inline src/add_remove_owner.test.js:42]: @dasanra says we can configure describe to fail all tests in a group.

**Comment by juampibermani** [comment]
Merging since it's passing all working tests.

**Pull request merged**